### ISSUE TITLE
Goto def on inlay hints doesn't distinguish sub types

### ISSUE THREAD

**Issue by rchiodo** [microsoft/pylance-release#6346]
Put code like so:

```python
from math import pi, tau
from sys import dllhandle


def foo(a: float, b: int, c: float):
    x: float = a + pi * dllhandle * tau * 0
    y: float = c
    z: int = b
    return x, y, z
```

Turn on inlay hints
Hover over the inlay hint
Goto def on tuple
Then try goto def on the first entry in the tuple `float`.

Expected result:
It jumps to the def for float

Actual result:
It goes to tuple still.

If you insert the hint, goto def on float works as expected, so it's just inlay hints that work differently.

**Comment by StellaHuang95** [comment]
This is expected. `Go to Definition` on inlay hints doesn't yet support `type arguments` of `class` types. Currently, there's no straightforward way to implement this due to the complexity of how `typePrinter` handles `Tuple` types. I'll mark this as an enhancement and consider supporting it if there's more ask from users.

Also, due to the way the VSCode UI displays underscores, it might look like that there's a separate underscore for `tuple` and `float`. However, the underscore is actually being partially obscured by taller characters, like the `y` in `Any`, which also cuts into the underscore. So, even though it looks like you're clicking on `float`, you're actually clicking on the entire `tuple[xxx]`.
![Image](https://github.com/user-attachments/assets/92a5cc93-79c3-409d-9eb2-c990b9fe53ee)



![Image](https://github.com/user-attachments/assets/bd6befa5-029d-49f8-b872-a8b0e263c8e7)

**Comment by rchiodo** [comment]
I don't think I'd classify this as an enhancement. Goto def when the code is actually there behaves differently than goto def on the inlay hint. To me that feels inconsistent, or a bug.

**Comment by StellaHuang95** [comment]
The intention is to support `Go to Definition` on the entire return type, not on every individual component within it. For example, in the function below, which returns a `GenericClass` type, the clickable link would be on the entire class type. We can talk in the triage if you really think it's a bug.

```python
from typing import TypeVar, Generic

T = TypeVar('T')

class GenericClass(Generic[T]):
    def __init__(self, value: T):
        self.value = value


def get_generic_instance() -> GenericClass[str]:
    return GenericClass(str())
```

**Comment by rchiodo** [comment]
I don't think it matters to the end user how it works, they just expect it to be consistent. IMO, anything that doesn't behave in a consistent way makes it feel off.

I think I'd argue for not shipping the original implementation until it's consistent with the type actually being there. That although the current implementation is useful, the inconsistency is worse overall.

**Comment by heejaechang** [comment]
I think we should ship the feature and see what feedback we get on it. I +1 on that this is enhancement. we already have a lot of inconsistency between features such as hover tooltip show one thing and completion tooltip show another and signature help tooltip something else, or literal type supported in dictionary, but not other context and etc. so far, we treated them as enhancement rather than bugs.

anyway, regardless of whether we mark it as bug or enhencement, I don't think it is ship blocker issue.

**Comment by StellaHuang95** [comment]
This issue has been fixed in prerelease version 2024.9.100, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#20249100-12-September-2024-PreRelease)
